pandas/core/internals/array_manager.py

```diff
@@ -480,31 +480,34 @@ def is_view(self) -> bool:
     def is_single_block(self) -> bool:
         return False
 
+    def _get_data_subset(self, predicate: Callable) -> ArrayManager:
```

Comment:
worth doing the same in BM?

Reply:
I would say no, because there it's a bit more complicated (needs to split object blocks to check for bool-like columns)